File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int left = 0;
        vector<int> counts(26, 0);
        int len = 0;
        int maxCount = 0;
        for (int right = 0; right < s.size(); right++) {
            maxCount = max(maxCount, ++counts[s[right] - 'A']);

            while(right - left + 1 - maxCount > k) {
                counts[s[left] - 'A']--;
                left++;
            }

            len = max(len, right - left + 1);
        }

        return len;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        counts = [0] * 26

        maxLen = 0

        left = 0

        for right in range(len(s)):
            counts[ord(s[right]) - ord('A')] += 1

            maxCount = max(counts)


            while right - left + 1 - maxCount > k:
                counts[ord(s[left]) - ord('A')] -= 1
                left += 1
                maxCount = max(counts)
                
            maxLen = max(maxLen, right - left + 1)

        return maxLen
            